pkg/kvstore/kvstore.go

```diff
@@ -35,88 +36,128 @@ const (
 	BaseKeyPrefix = "cilium"
 )
 
-// Get returns value of key
-func Get(key string) ([]byte, error) {
-	v, err := Client().Get(key)
+// ExtendedKVStore is a struct that export multiple kvstore methods
+type ExtendedKVStore struct {
+	namespace string
+	mutex     lock.Mutex
```

Comment:
This mutex only exists to protect changes to "namespace" (via SetNamespace()), but namespace is not ever changed. Maybe make namespace immutable and remove the mutex and SetNamespace(), and make GetNamespace() lock-free?